### PR TITLE
Add `remove-unused-repo-tabs` feature

### PR DESCRIPTION
**Pull request by yakov116**
Replaces and closes #4395

Test on
https://github.com/sindresorhus/eslint-plugin-unicorn 
https://github.com/sindresorhus/refined-github
https://github.com/pixiebrix/pixiebrix-extension
https://github.com/twschiller/dslfin-www/projects
https://github.com/twschiller/dslfin-www

image
image

WIP

### CODE REVIEW

**Comment by yakov116** [comment]
@fregante I did some research it looks like the best thing we can do here is display the counter always.

For example https://github.com/twschiller/dslfin-www has wiki enabled but only collaborators can edit. Any non collaborator gets redirected to the main repo page.

Hiding in such cases would not be good.

Can we change this feature to be `add-repo-tab-counters`?

**Comment by fregante** [comment]
> only collaborators can edit

The applies to every other tab we hide: projects, actions, wikis all can be edited by collaborators. 

For projects we added an item to the "+" dropdown; Should we do the same for the other 2?

**Comment by yakov116** [comment]
> For projects we added an item to the "+" dropdown; Should we do the same for the other 2?

I am 👎 on that, we are not "adding an action or wiki".

Should we make an API call to see if they are collaborators and share that across the features?

**Comment by yakov116** [comment]
What if we moved it to the "more drop down" when there are no actions/wiki?

**Comment by fregante** [comment]
> What if we moved it to the "more drop down" when there are no actions/wiki?

Not a bad idea. If it exists, append it there. If it doesn't, 🤷‍♂️.

**Comment by yakov116** [comment]
> > What if we moved it to the "more drop down" when there are no actions/wiki?
> 
> Not a bad idea. If it exists, append it there. If it doesn't, 🤷‍♂️.

@fregante what if they have `more-dropdown` disabled?

**Comment by shinenelson** [comment]
I came back to complete #4395 and this is already done \o/

**Review by shinenelson — COMMENTED** [review]
Should we, perhaps, consider renaming the `remove-projects-tab` to `add-projects-to-new-drop-down`?

Also, +1 to adding Actions to the `+` drop-down. Then we can re-purpose the existing `remove-actions-tab` to what it is currently doing `add-features-to-new-dropdown`.

**Comment by yakov116** [comment]
@fregante ?

**Comment by fregante** [comment]
> > If it doesn't, 🤷‍♂️.
> 
> @fregante what if they have `more-dropdown` disabled?

🤷‍♂️ => I don't care about them

**Comment by fregante** [comment]
> Should we, perhaps, consider renaming the `remove-projects-tab` to `add-projects-to-new-drop-down`?

That feature doesn't make sense anymore, it should be part of this one. The only reason to add a link there is because the tab has been removed.

**Comment by yakov116** [comment]
> That feature doesn't make sense anymore, it should be part of this one. The only reason to add a link there is because the tab has been removed.

Sorry I was out for a few days, I want to finish this up.

I have a few questions.

1. Are we keeping the "new project dropdown"?
2. If yes are we adding new actions or new wiki to it.

From what I understand we are not going to "remove" the tabs we are just going to move it to the `more-dropdown`

File: source/features/remove-unused-repo-tabs.tsx [hunk 1]
```diff
@@ -10,12 +10,16 @@ import looseParseInt from '../helpers/loose-parse-int';
...
+import {onlyShowInDropdown} from './more-dropdown';
 import {buildRepoURL, getRepo} from '../github-helpers';
 
-function tabCannotBeRemoved(tab: HTMLElement | undefined): boolean {
+async function tabCannotBeHidden(tab: HTMLElement | undefined): Promise<boolean> {
 	if (
 		!tab || // Tab disabled 🎉
-		tab.matches('.selected') // User is on tab 👀
+		tab.matches('.selected') ||// User is on tab 👀
+		// Repo/Organization owners should see the tab. If they don't need it, they should disable the tab altogether
```
Comment (shinenelson) [inline source/features/remove-unused-repo-tabs.tsx:20]:
```suggestion
		// Repo/Organization owners should see the tab. If they don't need it, they should disable the feature altogether
```

**Comment by yakov116** [comment]
@fregante can you give this a basic review? (As in  what this feature should include and how it should be split)

Also what should the screenshot be of?

**Comment by fregante** [comment]
Closing for cleanup. Open a new PR if you decide to pick it up

**Comment by yakov116** [comment]
I am waiting for your feedback...

**Comment by yakov116** [comment]
its fine I will open a new PR with all my questions

**Pull request closed without merging**